silkrpc/core/gas_price_oracle_test.cpp

```diff
@@ -0,0 +1,593 @@
+/*
+    Copyright 2021 The Silkrpc Authors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "gas_price_oracle.hpp"
+#include <boost/endian/conversion.hpp>
+
+#include <algorithm>
+#include <iostream>
+#include <catch2/catch.hpp>
+#include <asio/co_spawn.hpp>
+#include <asio/thread_pool.hpp>
+#include <asio/use_future.hpp>
+
+#include <evmc/evmc.hpp>
+
+#include <silkrpc/types/block.hpp>
+
+namespace silkrpc {
+
+using evmc::literals::operator""_address;
+
+static std::vector<silkworm::BlockWithHash> kBlocks;
```

Comment:
Please do not use global or module variables in unit tests: prefer the recommended rules from [Catch2](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#test-cases-and-sections) based on TEST_CASE and nested SECTIONs.

You should create this vector within each SECTION (as an aside this removes the need for `clear_blocks_vector` function), then call `fill_blocks_vector` with the vector instance as first parameter

Reply:
Done